guava/src/com/google/common/escape/UnicodeEscaper.java

```diff
@@ -127,14 +133,16 @@ public String escape(String string) {
    * @throws IllegalArgumentException if the scanned sub-sequence of {@code csq} contains invalid
    *     surrogate pairs
    */
-  protected int nextEscapeIndex(CharSequence csq, int start, int end) {
-    int index = start;
+  @SuppressWarnings("upperbound:compound.assignment.type.incompatible")/* (1) `Character.isSupplementaryCodePoint(cp)` is true when `cp` range from 65536 to 1114111.
+  Since `int cp = codePointAt(csq, index, end);` and `csq` is a sequence of characters. `cp` can only range from 97 to 122. */
```

Comment:
> `cp` can only range from 97 to 122.

Really? Why?

Reply:
Sorry I did not give an accurate description on this. 
`codePointAt()` return the Unicode code point for the given index or the negated value of the trailing high surrogate character at the end of the sequence. In `escapeSlow`, if cp is the negated value returned by `codePointAt()`, throw error. Else, return an integer between 0 and 65535 representing the UTF-16 code unit at the given index.